### PR TITLE
29 consume backend api

### PR DESCRIPTION
**Pull request by zaviermiller**
This PR creates a `services` and `types` folder. The `services/UserService.ts` provides a pretty good example of how we'll actually access the backend API from the frontend. Basically, we call the service methods which just abstract away the actual http requests. The `types/User.ts` defines an interface for the returned entity, and exports a blank object which can be used as a default.

Still need to figure out auth header for protected API routes -- #53

### CODE REVIEW

File: frontend/src/services/UserService.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const response = await http.get('/users');
+    return response.data;
+  }
+
+  static async getUser(id: number): Promise<User> {
+    const response = await http.get(`/users/${id}`);
+    return response.data;
+  }
+
+  static async createUser(user: User): Promise<User> {
```
Comment (andlrutt) [inline frontend/src/services/UserService.ts:15]: Does the promise type have to be user for all endpoints?

Comment (zaviermiller) [reply frontend/src/services/UserService.ts:15]: ggod point

File: frontend/src/services/UserService.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import http from '../util/http';
+import User from '../types/User';
+
+export default class UserService {
+  static async getUsers(): Promise<User[]> {
```
[the remark this answers is not in the document]
Comment (zaviermiller) [reply frontend/src/services/UserService.ts:5]: oh shoot yea there sure isnt, will remove

**Review by andlrutt — APPROVED** [review]
Fantastic!

**Pull request merged**